### ISSUE TITLE
Add a sphinx plugin to allow linking from demo configurations to api sections

### ISSUE THREAD

**Issue by phlax** [envoyproxy/envoy#15332]
### description

learning envoy configuration can be quite a steep learning curve and its not always easy to navigate the docs to find the relevant spec

im thinking it might be helpful if the config examples in the docs provided a click through to the relevant api section - perhaps if you hover over the line a link appears at the end, for example

depending on how fine grained the links were, i think we might need to add more ref links in the api to allow that

on a related point im noticing that all the api examples are `json` - whereas the config examples are mostly `yaml`. There are probably good reasons for both, but im thinking this is another way in which the docs can jar for new users.

**Comment by phlax** [comment]
cc @mattklein123 @htuch

**Comment by phlax** [comment]
cc @dmitri-d

**Comment by mattklein123** [comment]
Big +1 I think this would be amazing.

**Comment by dmitri-d** [comment]
@phlax: if we look at bootrstrap as an example, would you generate just one link pointing to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto, or also generate links for individual fields in addition to that? The latter is obviously more work, but should be doable.

**Comment by phlax** [comment]
> would you generate just one link pointing to https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/bootstrap/v3/bootstrap.proto, or also generate links for individual fields in addition to that? The latter is obviously more work, but should be doable.

yep, the latter is defo preferable, and yep i think it would require more refs to point to etc and add some complexity
